lib/u3d/log_analyzer.rb

```diff
@@ -189,12 +182,30 @@ def parse_line(line)
         end
       end
 
-      apply_ruleset.call(@phases[@active_phase]['rules'], @active_phase) if @active_phase
+      if @active_phase
+        apply_ruleset.call(@phases[@active_phase]['rules'], @active_phase)
+        if @phases[@active_phase]['phase_end_pattern'] && @phases[@active_phase]['phase_end_pattern'] =~ line
+          UI.verbose("--- Ending #{@active_phase} phase ---")
```

Comment:
do you want to move this UI.verbose under `finish_phase` ?

Reply:
I made the choice to take it out of it as this message is intended to tell when a rule end pattern has been matched.

But that could be confusing for the user using --verbose as he would see some rule ending messages for some and not for some others. What do you think about it?

Reply:
Ideally, I would like under verbose mode start and end messages to appear for rules and phases, and extra warnings when unexpected things happen (e.g. force aborting rules, etc.)

Reply:
OK. Logging rules start and end may end up generating lots of verbose though, basically logging twice or thrice as much.